aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/SubscriptionAuthorizer.java

```diff
@@ -160,24 +151,39 @@ private static JSONObject getAuthorizationDetailsForIAM(boolean connectionFlag,
         return authorizationMessage;
     }
 
-    private static JSONObject getAuthorizationDetailsForUserpools(
-            AWSConfiguration awsConfiguration, Context applicationContext) {
-        CognitoUserPool cognitoUserPool = new CognitoUserPool(applicationContext, awsConfiguration);
+    private JSONObject getAuthorizationDetailsForUserpools() {
+        CognitoUserPool cognitoUserPool = new CognitoUserPool(mApplicationContext, mAwsConfiguration);
         BasicCognitoUserPoolsAuthProvider basicCognitoUserPoolsAuthProvider = new BasicCognitoUserPoolsAuthProvider(cognitoUserPool);
         try {
             return new JSONObject()
-                .put("host", getHost(getApiUrl(awsConfiguration)))
+                .put("host", getHost(getApiUrl()))
                 .put("Authorization", basicCognitoUserPoolsAuthProvider.getLatestAuthToken());
         } catch (JSONException | MalformedURLException exception) {
             throw new RuntimeException("Error constructing authorization message JSON.", exception);
         }
     }
 
-    private static JSONObject getAuthorizationDetailsForOidc(
-            AWSConfiguration awsConfiguration, OidcAuthProvider mOidcAuthProvider) {
+    private AWSCredentialsProvider getCredentialsProvider() throws RuntimeException{
+        if (mCredentialsProvider != null) { return mCredentialsProvider; }
```

Comment:
Checks if `AWSCredentialsProvider` was already provided to us before creating a new one.